src/components/Archive/AllProducts/index.tsx

```diff
@@ -0,0 +1,44 @@
+import {useContext} from 'react';
+import ProductsContext from '../../../context/products-context';
+import CartContext from '../../../context/cart-context';
+
+import SingleProduct from '../SingleProduct';
+
+import { AllWrap } from './style';
+import { ProductObj, CartItem } from '../../../types/interfaces';
+
+const AllProducts = () => {
+	const productsCtx = useContext(ProductsContext);
+	const cartCtx = useContext(CartContext);
+
+	const onClickHandler = (e: React.MouseEvent<HTMLButtonElement> | React.TouchEvent<HTMLButtonElement>) => {
+		const id = parseInt(e.currentTarget.id);
+		const price = parseFloat(e.currentTarget.name);
+		let newCartItems;
+		//Getting product with setelcted id
+		const setItem = cartCtx.items.filter((item: CartItem) => item.id === id);
+		//If product is added for the forst time
+		if(setItem.length > 0){
+			const newQuantity = setItem[0]['quantity'] + 1;
+			const totalPrice = setItem[0]['total'] + price;
+			newCartItems = cartCtx.items.map((item: CartItem) => item.id === id ? { ...item, quantity: newQuantity, price: price, total: totalPrice} : item);
+		//If product is added for the first time
+		} else {
+			newCartItems = [...cartCtx.items, {id: id, quantity: 1, price: price, total: price}]
+		}
+		localStorage.setItem('items', JSON.stringify(newCartItems));
```

Comment:
if localStorage is disabled or there is error when requesting data from it, whole app will crash.
Find way to prevent this in all places where local storage is accessed.